Expensify/App
# Delete Comments

<If necessary, assign reviewers that know the area or changes well. Feel free to tag any additional reviewers you see fit.>

### Details
<Explanation of the change or anything fishy that is going on>

### Fixed Issues
Fixes https://github.com/Expensify/Expensify/issues/147483
Fixes https://github.com/Expensify/Expensify.cash/issues/2966

### Tests / QA Steps (web/desktop)
1. Hover over a message that you _did not_ write, and verify that you _do not_ see the `Edit Comment` or `Delete Comment` actions as options in the `ReportActionContextMenu`
1. Right-click a message that you _did not_ write, and verify that you _do not_ see the `Edit Comment` or `Delete Comment` actions as options in the `ReportActionContextMenu`
1. Hover over a message that you _did_ write, and press `Edit Comment`
1. Make a change to the message, and press `Save Changes`.
1. Verify that the change takes effect.
1. Wait for a few seconds, then refresh the page.
1. Verify that the comment remains edited.
1. Right-click a message that you _did_ write, and press `Edit Comment`
1. Make a change to the message, and press `Save Changes`.
1. Verify that the change takes effect.
1. Wait for a few seconds, then refresh the page.
1. Verify that the comment remains edited.
1. Hover over a message that you _did_ write, and press `Delete Comment`
1. A confirmation modal should appear. Press cancel.
1. Verify that the message is not deleted.
1. Wait for a few seconds, then refresh the page.
1. Verify that the message is still not deleted.
1. Hover over a message that you _did_ write, and press `Delete Comment`
1. A confirmation modal should appear. Press confirm.
1. Verify that the message no longer appears.
1. Wait for a few seconds, then refresh the page.
1. Verify that the message remains deleted.
1. Right-click a message that you _did_ write, and press `Delete Comment`
1. A confirmation modal should appear. Press confirm.
1. Verify that the message no longer appears.
1. Wait for a few seconds, then refresh the page.
1. Verify that the message remains deleted.
1. Hover over a message that you _did_ write, and press `Edit Comment`
1. Erase all the content of the message, then hit `Save Changes`.
1. Verify that the message no longer appears.
1. Wait for a few seconds, then refresh the page.
1. Verify that the message remains deleted.

### Tests / QA Steps (iOS/Android/mWeb)
1. LongPress a message that you _did not_ write, and verify that you _do not_ see the `Edit Comment` or `Delete Comment` actions as options in the `ReportActionContextMenu`
1. LongPress a message that you _did_ write, and press `Edit Comment`.
1. Make a change to the message and press `Save Changes`
1. Verify that the change takes effect.
1. (mWeb only) Wait for a few seconds, then refresh the page.
1. (mWeb only) Verify that the comment remains edited.
1. LongPress a message that you _did_ write, and press `Delete Comment`
1. A confirmation modal should appear. Press cancel.
1. Verify that the message _is not_ deleted.
1. (mWeb only) Wait for a few seconds, then refresh the page.
1. (mWeb only) Verify that the comment is still visible.
1. LongPress a message that you _did_ write, and press `Delete Comment`
1. A confirmation modal should appear. Press confirm.
1. Verify that the message no longer appears.
1. (mWeb only) Wait for a few seconds, then refresh the page.
1. (mWeb only) Verify that the comment remains deleted.
1. LongPress a message that you _did_ write, and press `Edit Comment`
1. Erase all the content of the message, then hit `Save Changes`
1. Verify that the message no longer appears.
1. (mWeb only) Wait for a few seconds, then refresh the page.
1. (mWeb only) Verify that the comment remains deleted.

### Tested On
